vslib/src/sai_vs_fdb.cpp

```diff
@@ -1,19 +1,281 @@
+
+        switch (attr.id)
+        {
+            case SAI_FDB_FLUSH_ATTR_BRIDGE_PORT_ID:
+
+                // remove from list all entries not matching bridge port id
+                if (fdb_attrs.at("SAI_FDB_ENTRY_ATTR_BRIDGE_PORT_ID")->getAttr()->value.oid != attr.value.oid)
+                {
+                    return true;
+                }
+
+                break;
+
+            case SAI_FDB_FLUSH_ATTR_VLAN_ID:
+
+                {
+                    sai_fdb_entry_t fdb_entry;
+
+                    sai_deserialize_fdb_entry(str_fdb_entry, fdb_entry);
+
+                    // remove from list all entries not matching vlan id
+                    if (fdb_entry.vlan_id != attr.value.u16)
+                    {
+                        return true;
+                    }
+                }
+
+                break;
+
+            case SAI_FDB_FLUSH_ATTR_ENTRY_TYPE:
+
+                // remove from list all entries not matching type
+                if (fdb_attrs.at("SAI_FDB_ENTRY_ATTR_TYPE")->getAttr()->value.s32 != attr.value.s32)
+                {
+                    return true;
+                }
+
+                break;
+
+            default:
+                SWSS_LOG_ERROR("flush attribute %d is not supported yet", attr.id);
+                return true;
+        }
+    }
+
+    // this fdb entry should be flushed
+    return false;
+}
+
 sai_status_t internal_vs_flush_fdb_entries(
         _In_ sai_object_id_t switch_id,
         _In_ uint32_t attr_count,
         _In_ const sai_attribute_t *attr_list)
 {
     SWSS_LOG_ENTER();
 
-    // TODO implement actual flush (9 cases) with ntf generation (queue)
-    // also update meta db here
+    /*
+     * There are 3 databases for fdb entries, one is in metadata and second is
+     * in g_fdb_info_set and third is local virtual switch. Second one holds
+     * all dynamic entries learned from interfaces. All learned entries are
+     * processed by metadata and propagated from info set to metadata. But
+     * there may be short period of time that those 3 sets will be out of sync
+     * untill notifications learned/aged will be sent.
+     *
+     * After learning fdb event, fdb entry will be put into 3 places:
+     * - local db g_switch_state_map
+     * - metadata db, and
+     * - g_fdb_info_set (contains only learned entries)
+     *
+     * This call should clear g_switch_state_map and g_fdb_info_set, and
+     * metadata db should be cleared by flush notification handler.
+     */
+
+    auto &fdbs = g_switch_state_map.at(switch_id)->objectHash.at(SAI_OBJECT_TYPE_FDB_ENTRY);
+
+    std::map<std::string, AttrHash> static_fdbs;
+    std::map<std::string, AttrHash> dynamic_fdbs;
+
+    for (auto it = fdbs.begin(); it != fdbs.end();)
+    {
+        if (doesFdbEntryNotMatchFlushAttr(it->first, it->second, attr_count, attr_list))
+        {
+            /*
+             * If fdb entry does not match flush attributes, erase it from all
+             * fdb entries.
```

Comment:
at the beginning we are getting all fdb entries present in the asic, and then we are filtering them, removing from list those which dont match input attributes, the list that will left will be flushed
since all input attributes acts as "and" operator

Reply:
it seems contradict the comments below " Also since we are using &on fdbs then this will also clear local data base". so, you remove those do not match from local data base?